### ISSUE TITLE
Request for configurations

### ISSUE THREAD

**Issue by ediadd** [NeuromorphicProcessorProject/snn_toolbox#60]
Hello,

I am currently using the snntoolbox to test SNN conversion accuracy under various configurations.
I was able to get useful information and really appreciate for providing this nice tool.

However, at some point, I found it is quite difficult to achieve error rates reported in the paper (Conversion of analog to spiking neural networks using sparse temporal coding).
For TTFS, I have tried out various configurations (duartions, dt, etc.) on Lenet-5, but I was unable to achieve the error rate of 2% (as reported in the paper).
The error rate is far worse than the error rate reported in the paper.
So, I was wondering if I'm making a mistake at some point (maybe the configuration was wrong).
If you don't mind, can I get the well-fit configuration files for temporal_mean_rate and TTFS coding scheme?

Thank you!

**Comment by rbodo** [comment]
Hi,

I just tried to run this myself again. Unfortunately, the old keras model I used for the paper does not work any more because of incompatibilities with more recent versions of keras/tf. So I had to train the model from scratch. After applying the TTFS conversion, I got a drop in accuracy of about 4.7 percentage points when going from ANN to SNN. That's clearly larger than the 2% reported in the paper, but not so large that I would suspect a bug. Might just be that the current training resulted in a larger proportion of low activations (i.e. late spikes), which are a source of error in the plain TTFS code.

Here's the configuration I used:

```
[paths]
path_wd = ./Data/snn_conversion/mnist/cnn/lenet5/keras/32bit
dataset_path = ./Datasets/mnist/cnn
filename_ann = mnist_cnn

[normalization]
percentile = 100

[conversion]
softmax_to_relu = True
spike_code = ttfs

[simulation]
duration = 50
dt = 0.1
num_to_test = 10000
batch_size = 50
```

The only important difference to a run with normal rate code is that we choose a higher time resolution (`dt = 0.1` instead of 1), because we rely on fine temporal precision here. Also, it may be beneficial to replace the softmax by a relu (`softmax_to_relu = True`) because softmax in TTFS can only approximate the ANN softmax. ReLU is a safe choice as long as not all output neurons have a negative pre-activation (in which case there would be no spikes even though the softmax would be non-zero).

If the configuration above does not close the gap sufficiently, you may want to try one of the TTFS code variants or train the network with low activations clamped to zero (which should give the best results).

For an example of a temporal_mean_rate config, see [here](https://github.com/NeuromorphicProcessorProject/snn_toolbox/blob/master/examples/mnist_keras_INI.py).
